### PR TITLE
this adds sorting and visible stats to the instructor tiles

### PR DESCRIPTION
**Pull request by joelhooks**
This adds sorting to the instructor pulse list so that they are descending in a more logical order based on what needs attention. This also adds some visible stats to the individual cards to get a feel for how they are doing at a glance.

All very basic, but moving in the right direction.

### CODE REVIEW

File: src/screens/Admin/screens/InstructorPulse/components/InstructorsList/components/InstructorSummary/index.js
Comment (joelhooks) [inline src/screens/Admin/screens/InstructorPulse/components/InstructorsList/components/InstructorSummary/index.js:7]: We cool with small things like this inline as a const? Doesn't seem like it deserves a file.

File: src/screens/Admin/screens/InstructorPulse/index.js
Comment (joelhooks) [inline src/screens/Admin/screens/InstructorPulse/index.js:16]: these lines are where the magic happens. *[sort-by](https://www.npmjs.com/package/sort-by)* seemed reasonable for the task.

1. Are they waiting for approval?
2. Have they claimed a lessons?
3. Do they have an approved lesson?
4. Newest first.

File: src/screens/Admin/screens/InstructorPulse/index.js
Comment (joelhooks) [inline src/screens/Admin/screens/InstructorPulse/index.js]: Published lesson count is the only one that sorts ascending, so people with the *least* number of published lessons are closer to the top of the list. Basically, I will **always** expect @johnlindquist to be at the bottom of this list. He needs the least help.

**Comment by joelhooks** [comment]
The required API changes have been deployed to @eggheadio/egghead-rails. 

FYI this api is slow for admins because of the revenue calculation for all instructors. We could massively improve it by providing an `instructor_revenue_url` to load async.

File: package.json
Comment (trevordmiller) [inline package.json:46]: Could use lodash so the extra dep isn't needed but nbd; https://lodash.com/docs/4.17.4#sortBy

File: src/screens/Admin/screens/InstructorPulse/components/InstructorsList/components/InstructorSummary/index.js
Comment (trevordmiller) [reply src/screens/Admin/screens/InstructorPulse/components/InstructorsList/components/InstructorSummary/index.js:7]: Yeah I'm cool with whatever makes sense :)

File: src/screens/Admin/screens/InstructorPulse/index.js
Comment (trevordmiller) [reply src/screens/Admin/screens/InstructorPulse/index.js:16]: Ah, ok I see why you used this now instead of lodash. Cool, looks good! And this will be fine to move this view into the root and allow all to see it because this data is only returned from the response if the instructor has access to it, so there is no risk of other instructors seeing royalty numbers and stuff like that, right?

Comment (trevordmiller) [reply src/screens/Admin/screens/InstructorPulse/index.js:16]: In the future, might also be good to have a `joined` property in the future or something that lets us know how new the instructor is so we can put new (maybe joined in the last three months or something) instructors at the top regardless of their lesson states?

**Review by trevordmiller — APPROVED** [review]
Looks great. Thanks @joelhooks! Approving and merging.

File: package.json
Comment (trevordmiller) [reply package.json:46]: Nevermind, I see why now :)

**Comment by trevordmiller** [comment]
> "We could massively improve it by providing an instructor_revenue_url to load async."

yeah that would be nice.

File: src/screens/Admin/screens/InstructorPulse/index.js
Comment (joelhooks) [reply src/screens/Admin/screens/InstructorPulse/index.js:16]: They are actually sorted by ID which correlates to join date fairly nicely. I think that "currently waiting for approval" takes priority over "hasn't submitted" which is why I did it this way. Ultimately I'd like live filters that I could configure.

Comment (trevordmiller) [reply src/screens/Admin/screens/InstructorPulse/index.js:16]: Sounds good. Thanks again for the PR.

**Pull request merged**
